kotlinx-coroutines-core/common/src/CoroutineScope.kt

```diff
@@ -10,102 +10,481 @@ import kotlin.coroutines.*
+ * The [CoroutineScope.cancel] extension function shall be used when the entity launching coroutines
+ * is no longer needed. It cancels all the coroutines that might still be running on its behalf.
+ *
+ * ```
+ * class MyEntity(scope: CoroutineScope? = null): AutoCloseable {
+ *    // careful: do not write `get() =` here by accident!
+ *    private val scope = scope ?: CoroutineScope(SupervisorJob() + CoroutineExceptionHandler { _, e ->
+ *        println("Error in coroutine: $e")
+ *    })
+ *
+ *    fun doSomethingWhileEntityExists() = scope.launch {
+ *        while (true) {
+ *            // do some work
+ *            delay(50.milliseconds)
+ *            println("Doing something")
+ *        }
+ *    }
+ *
+ *    override fun close() {
+ *        // cancel all computations related to this entity
+ *        scope.cancel()
+ *    }
+ * }
+ *
+ * fun main() {
+ *     MyEntity().use { entity ->
+ *         entity.doSomethingWhileEntityExists()
+ *         Thread.sleep(200)
+ *         entity.close()
+ *     }
+ * }
+ * ```
+ *
+ * Usually, a custom [CoroutineScope] should be created with a [SupervisorJob] and
+ * a [CoroutineExceptionHandler] to handle exceptions in child coroutines.
+ * See the documentation for the [CoroutineScope] constructor function for more details.
+ * Also note that `MyEntity` accepts the `scope` parameter that can be used to pass a custom scope for testing.
+ *
+ * Sometimes, authors of coroutine-aware frameworks provide [CoroutineScope] instances like this out of the box.
+ * For example, on Android, all entities with a lifecycle and all `ViewModel` instances expose a [CoroutineScope]:
+ * see [the corresponding documentation](https://developer.android.com/topic/libraries/architecture/coroutines).
+ *
+ * ### Taking another view of an existing scope
+ *
+ * Occasionally, several coroutines need to be launched with the same additional [CoroutineContext] that is not
+ * present in the original scope.
+ * In this case, the [CoroutineScope.plus] operator can be used to create a new view of the existing scope:
+ *
+ * ```
+ * coroutineScope {
+ *     val sameScopeButInUiThread = this + Dispatchers.Main
+ *     sameScopeButInUiThread.launch {
+ *         println("Running on the main thread")
+ *     }
+ *     launch {
+ *         println("This will run using the original dispatcher")
+ *     }
+ *     sameScopeButInUiThread.launch {
+ *         println("And this will also run on the main thread")
+ *     }
+ * }
+ * ```
+ *
+ * The lifecycle of the new scope is the same as the original one, but the context includes new elements.
+ *
+ * ### Application lifecycle scope
+ *
+ * [GlobalScope] is a [CoroutineScope] that has the lifetime of the whole application.
+ * It is convenient for launching top-level coroutines that are not tied to the lifecycle of any entity,
+ * but it is easy to misuse it and create memory leaks or resource leaks when a coroutine actually should be tied
+ * to the lifecycle of some entity.
```

Comment:
Splitting removes the focus on the contradiction, which is the whole point. Would this work instead?
```suggestion
 * Although is convenient for launching top-level coroutines that are not tied to the lifecycle of any entity,
 * it is easy to misuse it and create memory leaks or resource leaks when a coroutine actually should be tied
 * to the lifecycle of some entity.
```